include/boost/geometry/algorithms/append.hpp

```diff
@@ -91,19 +91,22 @@ template <typename Polygon, typename Point>
 struct point_to_polygon
 {
     typedef typename ring_type<Polygon>::type ring_type;
+	typedef typename interior_return_type<Polygon>::type interior_ring_range_type;
 
     static inline void apply(Polygon& polygon, Point const& point,
                 int ring_index, int = 0)
     {
         if (ring_index == -1)
         {
+			ring_type& ext_ring = exterior_ring(polygon);
```

Comment:
`exterior_ring()` may return not-real reference so you should use `ring_return_type<Polygon>::type` here, similar to `interior_return_type<Polygon>::type` you used. Otherwise you may end up with dangling reference,